patterns/1-initial/innersource-before-open-source.md

```diff
@@ -0,0 +1,73 @@
+
+## Story
+
+A large tech company once open-sourced a widely used internal tool, expecting external developers to contribute immediately. However, due to a lack of contributor guidelines, onboarding processes, and structured governance, external adoption was low, and internal maintainers were overwhelmed with unstructured contributions and support requests.
+
+After seeing this, the company implemented an InnerSource-first policy, ensuring internal teams could practice open collaboration before releasing future projects as open source.  
+
+## Context
+
+This pattern applies in organizations that:
+
+- Want to release internal software as open source.  
+- Lack structured internal collaboration processes.  
+- Have teams unfamiliar with maintaining open source projects.  
+- Need to establish internal governance and contribution models before engaging the broader open source community.  
+
+## Forces
+
+- **Collaboration Readiness**: Teams may not be used to handling external contributions or asynchronous collaboration.  
+- **Documentation Gaps**: A lack of contributor guidelines, API documentation, and onboarding materials can hinder adoption.  
+- **Governance & Ownership**: Without clear ownership and decision-making processes, project direction can become unclear.  
+- **Support Burden**: Open source projects require active maintainers to review pull requests, address issues, and engage the community.  
+- **Security & Compliance**: Code may require review to meet licensing and security requirements before being released publicly.  
+
+## Solution
+
+Before making a project open source, require it to go through an InnerSource phase where:
+
+1. The project is made available internally for contributions from other teams.  
+2. Clear documentation, contribution guidelines, and governance structures are established.  
+3. Maintainers gain experience managing contributions, reviewing pull requests, and addressing issues.  
+4. Internal adoption and success metrics are measured to determine if the project is ready for external release.  
+5. Feedback loops are created to refine processes before engaging a broader open source audience.  
+
+## Resulting Context
+
+- Teams develop the skills necessary to manage open source projects effectively.  
+- Contributor documentation and governance structures are established and tested.  
+- Internal adoption increases, providing validation of the project's value before external release.  
+- The transition to open source is smoother, with better preparedness for external collaboration.  
+
+## Rationale
+
+By requiring InnerSource before Open Source, organizations ensure that projects are equipped with the right practices and infrastructure to thrive in an open community. This approach mitigates risks, improves sustainability, and maximizes the chances of long-term success.  
+
+## Known Instances
+
+TBD
```

Comment:
Would you like to add your org as a known instance here?

Reply:
@fer-correa would you like to add your org here as a known instance? If you don't feel comfortable doing that yet, or have to get some sort of approval, I totally understand.